File: dotnet/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example1/Follower.cs
namespace ObserverDesignPattern.Example1;

public class Follower : IObserver<Article>
{
    public string Email { get; }
    public string Name { get; }

    public Follower(string name, string email)
    {
        Name = name;
        Email = email;
    }

    public void Update(Article article)
    {
        Console.WriteLine($"Email sent to {Email}: '{article.Title}' by author at {article.PublishedAt}");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example1/src/Follower.ts
import { IObserver } from './IObserver';
import { Article } from './Article';

export class Follower implements IObserver<Article> {
  readonly name: string;
  readonly email: string;

  constructor(name: string, email: string) {
    this.name = name;
    this.email = email;
  }

  update(article: Article): void {
    console.log(`Email sent to ${this.email}: '${article.title}' by author at ${article.publishedAt.toISOString()}`);
  }
}
